api/rql/internal/nonterminalNode.go

```diff
@@ -0,0 +1,64 @@
+package internal
+
+import "github.com/puppetlabs/wash/api/rql"
+
+import "github.com/puppetlabs/wash/api/rql/internal/errz"
+
+// A NonterminalNode is a node that can be matched by one or
+// more other nodes when it is unmarshaled. MatchedNode returns
+// the matched node
+type NonterminalNode interface {
+	rql.ASTNode
+	MatchedNode() rql.ASTNode
+	SetMatchedNode(rql.ASTNode) NonterminalNode
+	SetMatchErrMsg(msg string) NonterminalNode
+}
+
+func NewNonterminalNode(n rql.ASTNode, ns ...rql.ASTNode) NonterminalNode {
```

Comment:
Why two arguments? I guess that makes it so you need at least 1?

Reply:
Yup. I'll comment that.

Reply:
Updated